backend/routes.js

```diff
@@ -36,28 +46,55 @@ const route = new Proxy({}, {
 //       —BUT HTTP2 might be better than websockets and so we keep this around.
 //       See related TODO in pubsub.js and the reddit discussion link.
 route.POST('/event', {
-  auth: 'gi-auth',
+  auth: {
+    strategy: 'gi-auth',
+    mode: 'optional'
+  },
   validate: { payload: Joi.string().required() }
 }, async function (request, h) {
+  // TODO: Update this regex once `chel` uses prefixed manifests
+  const manifestRegex = /^z9brRu3V[123456789ABCDEFGHJKLMNPQRSTUVWXYZabcdefghijkmnopqrstuvwxyz]{44}$/
   try {
     console.debug('/event handler')
     const deserializedHEAD = GIMessage.deserializeHEAD(request.payload)
     try {
+      if (!manifestRegex.test(deserializedHEAD.head.manifest)) {
+        return Boom.badData('Invalid manifest')
+      }
+      const credentials = request.auth.credentials
+      // Only allow identity contracts to be created without attribution
+      if (!credentials?.billableContractID && deserializedHEAD.contractID === deserializedHEAD.hash) {
```

Comment:
Maybe this is a good time to define `.isFirstMessage()` for the object returned by `deserializeHEAD`? I think that would be clearer than this.